lib/repl.js

```diff
@@ -1141,6 +1141,165 @@ REPLServer.prototype.convertToContext = function(cmd) {
   return cmd;
 };
 
+REPLServer.prototype.setupHistory = function(historyPath, oldHistoryPath, cb) {
```

Comment:
At very least, this would have to be refactored so that only the internal one uses `oldHistoryPath` -- no good in exposing deprecated functionality. :)

Another question this bears is, does the repl try to do this when you create it, or do you have to tell it to? The later potentially has a race condition.